openslides/motions/static/js/motions/site.js

```diff
@@ -1607,73 +1607,63 @@ angular.module('OpenSlidesApp.motions.site', [
     'Category',
     'Motion',
     'User',
-    function($scope, $q, gettext, Category, Motion, User) {
+    'gettextCatalog',
+    function($scope, $q, gettext, Category, Motion, User, gettextCatalog) {
         // set initial data for csv import
         $scope.motions = [];
-        $scope.separator = ',';
-        $scope.encoding = 'UTF-8';
-        $scope.encodingOptions = ['UTF-8', 'ISO-8859-1'];
-        $scope.accept = '.csv, .txt';
-        $scope.csv = {
-            content: null,
-            header: true,
-            headerVisible: false,
-            separator: $scope.separator,
-            separatorVisible: false,
-            encoding: $scope.encoding,
-            encodingVisible: false,
-            accept: $scope.accept,
-            result: null
-        };
-        // set csv file encoding
-        $scope.setEncoding = function () {
-            $scope.csv.encoding = $scope.encoding;
-        };
-        // set csv file encoding
-        $scope.setSeparator = function () {
-            $scope.csv.separator = $scope.separator;
-        };
-        // detect if csv file is loaded
-        $scope.$watch('csv.result', function () {
+
+        // set csv 
+        $scope.csvConfig = {
+            accept: '.csv, .txt',
+            encodingOptions: ['UTF-8', 'ISO-8859-1'],
+            parseConfig: {
+                skipEmptyLines: true,
+            },
+        };
+
+        var FIELDS = ['identifier', 'title', 'text', 'reason', 'submitter', 'category', 'origin'];
+        $scope.motions = [];
+        $scope.onCsvChange = function (csv) {
             $scope.motions = [];
-            var quotionRe = /^"(.*)"$/;
-            angular.forEach($scope.csv.result, function (motion) {
-                if (motion.identifier) {
-                    motion.identifier = motion.identifier.replace(quotionRe, '$1');
-                    if (motion.identifier !== '') {
-                        // All motion objects are already loaded via the resolve statement from ui-router.
-                        var motions = Motion.getAll();
-                        if (_.find(motions, function (item) {
-                            return item.identifier == motion.identifier;
-                        })) {
-                            motion.importerror = true;
-                            motion.identifier_error = gettext('Error: Identifier already exists.');
-                        }
+            if (!csv) {
+                return;
```

Comment:
I am not a friend of calling return deep inside a function. So we may find a solution where this is not necessary? If csv is undefined then the following forEach loops may not run and let the motions array empty. Or you find a positive call (if csv is not undefind, do this or that ...).